packages/common/integration/tests/directives/amount-format.spec.ts

```diff
@@ -1,113 +1,1104 @@
+        let control: Partial<NgControl>;
 
-        const element: Partial<ElementRef<Partial<HTMLInputElement>>> = {
-            nativeElement: { value: '' }
-        };
+        beforeEach(() => {
+            ngModelValue = '';
 
-        const control: Partial<NgControl> = {
-            reset(value?: string | number): void {
-                ngModelValue = value;
-            }
-        };
-
-        const directive = new AmountFormatDirective(element as ElementRef, control as NgControl);
-
-        directive.format();
-        expect(element.nativeElement!.value).toEqual('');
-        expect(ngModelValue).toEqual('');
-
-        element.nativeElement!.value = '15000';
-        directive.format();
-        expect(element.nativeElement!.value).toEqual('15 000');
-        expect(ngModelValue).toEqual(15000);
-
-        element.nativeElement!.value = '-100,000';
-        directive.format();
-        expect(element.nativeElement!.value).toEqual('-100');
-        expect(ngModelValue).toEqual(-100);
-
-        element.nativeElement!.value = '-100.100';
-        directive.format();
-        expect(element.nativeElement!.value).toEqual('-100.1');
-        expect(ngModelValue).toEqual(-100.1);
-
-        element.nativeElement!.value = '199900,02000';
-        directive.format();
-        expect(element.nativeElement!.value).toEqual('199 900.02');
-        expect(ngModelValue).toEqual(199900.02);
-
-        element.nativeElement!.value = '9123.50';
-        directive.format();
-        expect(element.nativeElement!.value).toEqual('9 123.5');
-        expect(ngModelValue).toEqual(9123.5);
-
-        element.nativeElement!.value = '100,200,300';
-        directive.format();
-        expect(element.nativeElement!.value).toEqual('100.2');
-        expect(ngModelValue).toEqual(100.2);
-
-        element.nativeElement!.value = '-10000000000.0009';
-        directive.format();
-        expect(element.nativeElement!.value).toEqual('-10 000 000 000.001');
-        expect(ngModelValue).toEqual(-10000000000.001);
-
-        element.nativeElement!.value = 'ASD';
-        directive.format();
-        expect(element.nativeElement!.value).toEqual('');
-        expect(ngModelValue).toEqual('');
+            element = {
+                nativeElement: {
+                    value: '',
+                    setSelectionRange() {}
+                }
+            };
+
+            control = {
+                value: null,
+                reset(value?: string | number): void {
+                    // @ts-ignore
+                    control['value'] = ngModelValue = value;
+                }
+            };
+
+            // MOCK
+            directive = new AmountFormatDirective(element as ElementRef, DEFAULT_AMOUNT_OPTIONS, control as NgControl);
+        });
+
+        describe('ru-RU', () => {
+            describe('non numeric by ru-RU', () => {
+                it('ngModel value is empty by ru-RU', () => {
+                    directive.format();
```

Comment:
seriously, it's a crazy big job to match it all

Reply:
about syntax: you deliberately avoided the `test.each` construct?

Reply:
`about syntax: you deliberately avoided the test.each construct?` 

yes, for better readability of tests